Code/Clavier/main/AnalogKey.cpp

```diff
@@ -12,34 +20,27 @@ AnalogKey::~AnalogKey(){
 
 }
 
-void AnalogKey::SetMacro(const int (&macro)[MAX_MACRO_SIZE]) {
-
-  for(auto i = 0; i < MAX_MACRO_SIZE; i++){
-    _macro[i] = macro[i];
-    Serial.println(_macro[i]);
-  }
+const int AnalogKey::GetValue(){
+  return _analogValue;
 
 }
 
-void AnalogKey::SetJoystick(enum Joystick direction){
+void AnalogKey::SetJoystick(enum JoystickPos direction){
 
   _direction = direction;
 
 }
 
-unsigned int AnalogKey::BuffAvg(){
-
-  auto avg = 0u;
-
-  for( int i =0 ; i<BUFFER_SIZE ; i++){
-    avg += _buffer[i];
+void AnalogKey::SetMacro(const int (&macro)[MAX_MACRO_SIZE]) {
 
+  for(auto i = 0; i < MAX_MACRO_SIZE; i++){
```

Comment:
Idéalement, tu utiliserais un range for loop : for(const auto& mac : macro) si ça fonctionne